src/v/redpanda/admin_server.cc

```diff
@@ -349,19 +349,89 @@ ss::future<ss::httpd::redirect_exception> admin_server::redirect_to_leader(
             leader_id_opt.value()),
           ss::httpd::reply::status_type::service_unavailable);
     }
+    auto leader = leader_opt.value();
+
+    // Heuristic for finding peer's admin API interface that is accessible
+    // from the client that sent this request:
+    // - if the host in the Host header matches one of our advertised kafka
+    //   addresses, then assume that the peer's advertised kafka address
+    //   with the same index will also be their public admin API address.
+    // - Assume that the peer is listening on the same port that the client
+    //   used to make this request (i.e. the port in Host)
+    //
+    // This will work reliably if all node configs have symmetric kafka listener
+    // sections (i.e. all specify the same number of listeners in the same
+    // order, for example all nodes have an internal and an external listener in
+    // that order), and the hostname used for connecting to the admin API
+    // matches one of the hostnames used for a kafka listener.
+    //
+    // The generic fallback if the heuristic fails is to use the peer's
+    // internal RPC address.  This works if the user is e.g. connecting
+    // by IP address to a k8s cluster's internal pod IP.
+
+    auto host_hdr = req.get_header("host");
```

Comment:
I assume this is case insensitive getter?

Reply:
Yes, the map definition in seastar request uses a case insensitive definition